### PR TITLE
patches done and added updated Jenkins file

### PR DESCRIPTION
**Pull request by HariKallam**
Did the following

1. Added Jenkins file as its missing and patched it from pull/8  for new changes
2. Did the first four patches from the Pull/8

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Thanks for your work! A few items

1. Please tag this PR appropriately 
2. Please update the Jenkins file with the latest groovy version. You have the older Jenkinsfile
3. Please setup your .gitconfig locally, and amend your commit. You do not have your username or email correct in your .gitconfig

CI is not passing as Jenkins is missing some permissions.

**Comment by chrislovecnm** [comment]
test this please

File: scripts/validate.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# limitations under the License.
+
+# bash "strict-mode", fail immediately if there is a problem
+set -o nounset
+set -o pipefail
+
+SUCCESS="hello-server-"
+FAILURE="Error from server (Forbidden)"
+PODLABELER="pod-labeler-"
+UPDATED="updated="
```
Comment (geojaz) [inline scripts/validate.sh]: I think shellcheck is dying because you're not using UPDATED. I think you can remove this variable.

**Comment by chrislovecnm** [comment]
It you look at https://github.com/GoogleCloudPlatform/gke-rbac-demo/commit/6640072dbd0c409d4d2d7c5ea0bd6de66b50fb92.patch it shows the incorrect email address.  I am unable to commit this code with the commit this way :)

**Comment by HariKallam** [comment]
> It you look at https://github.com/GoogleCloudPlatform/gke-rbac-demo/commit/6640072dbd0c409d4d2d7c5ea0bd6de66b50fb92.patch it shows the incorrect email address. I am unable to commit this code with the commit this way :)

I corrected the sha author issue. Pls see if its good to go

File: scripts/validate.sh (hunk 1 above)
Comment (HariKallam) [reply scripts/validate.sh]: I updated the above change in this PR, Thanks

**Pull request merged**